airbyte-cdk/python/airbyte_cdk/sources/message/repository.py

```diff
@@ -0,0 +1,36 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from abc import ABC, abstractmethod
+from typing import Iterable
+
+from airbyte_cdk.models import AirbyteMessage, Type
+
+
+class MessageRepository(ABC):
+    @abstractmethod
+    def emit_message(self, message: AirbyteMessage) -> None:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def consume_queue(self) -> Iterable[AirbyteMessage]:
+        raise NotImplementedError()
+
+
+class InMemoryMessageRepository(MessageRepository):
```

Comment:
Random thought: this should probably be a singleton; it would be pretty cool to not have to instantiate and pass this through the whole CDK. However, there are threading implications to do this and therefore I think we should delay this